features/netsocket/PPPInterface.h

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef PPP_INTERFACE_H
+#define PPP_INTERFACE_H
+
+#include "nsapi.h"
+#include "OnboardNetworkStack.h"
+#include "NetworkInterface.h"
+#include "PPP.h"
+
+
+/** PPPInterface class
+ *  Implementation of the NetworkInterface for an PPP-service
+ *
+ * This class provides the necessary glue logic to create a NetworkInterface
+ * based on an PPP and an OnboardNetworkStack.
+ *
+ */
+class PPPInterface : public virtual NetworkInterface {
+public:
+    /** Create an PPP-based network interface.
+     *
+     * The default arguments obtain the default PPP, which will be target-
+     * dependent (and the target may have some JSON option to choose which
+     * is the default, if there are multiple). The default stack is configured
+     * by JSON option nsapi.default-stack.
+     *
+     * Due to inability to return errors from the constructor, no real
+     * work is done until the first call to connect().
+     *
+     * @param ppp  Reference to PPP to use
+     * @param stack Reference to onboard-network stack to use
+     */
+    PPPInterface(PPP &ppp = PPP::get_default_instance(),
+                 OnboardNetworkStack &stack = OnboardNetworkStack::get_default_instance());
+    virtual ~PPPInterface();
+
+    /** Set a static IP address
+     *
+     *  Configures this network interface to use a static IP address.
+     *  Implicitly disables DHCP, which can be enabled in set_dhcp.
+     *  Requires that the network is disconnected.
+     *
+     *  @param ip_address  Null-terminated representation of the local IP address
+     *  @param netmask     Null-terminated representation of the local network mask
+     *  @param gateway     Null-terminated representation of the local gateway
+     *  @return            0 on success, negative error code on failure
+     */
+    virtual nsapi_error_t set_network(const char *ip_address, const char *netmask, const char *gateway);
+
+    /** Enable or disable DHCP on the network
+     *
+     *  Requires that the network is disconnected
+     *
+     *  @param dhcp     False to disable dhcp (defaults to enabled)
```

Comment:
Does this actually have a default argument?

Reply:
Removed whole function from PPP interface since dhcp is not valid for ppp. It is set false on stack bring up.